### ISSUE TITLE
containerSrc could be long (a data URI)

### ISSUE THREAD

**Issue by tdresser** [w3c/longtasks#64]
We should likely truncate containerSrc in cases where it's too long.

**Comment by domenic** [comment]
We don't do this for anything else on the platform that exposes a URL... what's the rationale?

**Comment by tdresser** [comment]
If you completely delete an element, it's associated PerformanceEntry could keep the full data URI alive for no good reason. I believe this is different from other ways URLs are exposed today.

This may just be premature optimization though.

@yoavweiss @rniwa @toddreifsteck, any thoughts?

**Comment by toddreifsteck** [comment]
My thought is that this is premature optimization. @rniwa ?

**Comment by hawkinsw** [comment]
> If you completely delete an element, it's associated PerformanceEntry could keep the full data URI alive for no good reason. I believe this is different from other ways URLs are exposed today.

Can you explain to me why the PerformanceEntry would stay alive for an element that is completely deleted? I am sorry to be stupid.

> 
> This may just be premature optimization though.
> 
> @yoavweiss @rniwa @toddreifsteck, any thoughts?

**Comment by rniwa** [comment]
I mean... there is no mechanism for PerformanceEntry to removed for long tasks right now, right?

**Comment by hawkinsw** [comment]
At this point then, what we are saying is that an implementer might choose
to keep it around because there is nothing in the spec that mandates its
deletion? In such a case, a long URL could have a negative impact on memory
consumption of the program?

**Comment by toddreifsteck** [comment]
The PerformanceEntry is enqueued to Performanee Timeline. Is the underlying question whether the Performance Timeline can delete the entry if there is no buffering behavior specified?

**Comment by rniwa** [comment]
> At this point then, what we are saying is that an implementer might choose to keep it around because there is nothing in the spec that mandates its deletion

It's not so much that an implementation may choose to keep it in the timeline. The specifications mandates that whatever entry added to the timeline remain in the timeline unless the buffering limit is reached, or it's explicitly cleared. Whether the associated element is removed from the document is irrelevant in this regard.

**Comment by hawkinsw** [comment]
I think that's what I was asking, Todd! Thanks for being accurate.

I think that Ryosuke answered my question.

I do wonder, though, why we did not address removing Performance Entries
from the Performance Timeline if/when their associated
elements are deleted from the document.

**Comment by yoavweiss** [comment]
> The PerformanceEntry is enqueued to Performance Timeline. Is the underlying question whether the Performance Timeline can delete the entry if there is no buffering behavior specified?

IIRC, currently for LongTasks, they are only queued for a PerformanceObserver, which means that a site that ignores those entries will not keep that URL alive. We are talking about keeping them in the timeline in the future though (through explicit registration).

It may make sense to explicitly set apart data URIs from e.g. HTTP schemes.

It's also worth noting that at least in some implementations, those URLs can take up that space exactly once, even if multiple LongTask entries point at them.




> I do wonder, though, why we did not address removing Performance Entries from the Performance Timeline if/when their associated elements are deleted from the document.

IMO, it doesn't really make sense to remove such entries from the timeline, as they did happen (and impacted the user). Later removal of the element doesn't change that.

**Comment by npm1** [comment]
Yes, right now there is no buffered flag behavior nor performance timeline buffering for longtask entries, so they would only need to be kept alive if a PerformanceObserver looking at them stores them in JS. Is it common to have long inline scripts? Exposing the full URI does not seem to be flagged to Chrome as a memory issue thus far. Maybe it could become one if we support buffering longtask entries in the future.

**Comment by toddreifsteck** [comment]
Per call on 8/1: Consider an arbitrary limit for data uri's universally for all Performance Timeline exposed PerformanceEntry's.

**Comment by yoavweiss** [comment]
@npm1 - I know that we're applying limits on URL length elsewhere. Would be good to consolidate our approach.

**Comment by npm1** [comment]
Yes, although for different resource types. Consolidating still seems reasonable to me.

**Comment by npm1** [comment]
Similar issue: https://github.com/WICG/element-timing/issues/64

**Comment by noamr** [comment]
I wonder if we should follow [this guideline](https://infra.spec.whatwg.org/#algorithm-limits), and leave this limitation implementation-defined.
